lib/web_ui/lib/pointer.dart

```diff
@@ -153,7 +153,23 @@ class PointerData {
   }
 }
 
+/// A function used to notify the engine about how a [PointerDataPacket] was used in the framework.
+///
+/// For example, can be used to [preventDefault] the source events on the web.
+typedef _PointerDataPacketAcknowledgeFn = void Function({
+  bool? preventDefault,
+});
+
+// The default _PointerDataPacketAcknowledgeFn, which is a noop.
+void _defaultPointerDataPacketAcknowledgeFn ({
+  bool? preventDefault,
+}) {}
+
 class PointerDataPacket {
-  const PointerDataPacket({this.data = const <PointerData>[]});
+  const PointerDataPacket({
+    this.data = const <PointerData>[],
+    this.acknowledge = _defaultPointerDataPacketAcknowledgeFn,
+  });
   final List<PointerData> data;
+  final _PointerDataPacketAcknowledgeFn acknowledge;
```

Comment:
Also, it is odd that this is on the `PointerDataPackage` (which could contain multiple unrelated events), but from the framework side at least it appears that we want to target individual events.

Reply:
I think this is the right granularity. The Framework has no idea about the underlying native events; all it sees is a "data packet".

My idea is that the engine implementation of the `acknowledge` function (see the `_sendToFramework` below) can be made as complex as needed to handle the list of data packets individually (if it is ever needed).